tests/dummy/app/states/todos.js

```diff
@@ -1,6 +1,8 @@
 import BufferedProxy from 'ember-buffered-proxy/proxy';
 
-BufferedProxy.reopenClass({
+const TodosProxy = BufferedProxy.extend();
+
+TodosProxy.reopenClass({
   initialState() {
     return {
       content: this.get('todo')
```

Comment:
@blimmer the clobbering that you are talking about would not happen as the only thing initialState is used for is producing an initialState object (https://github.com/stefanpenner/ember-state-services/blob/master/addon/state-for.js#L72-L74) and then that object is used to set the defaults here (https://github.com/stefanpenner/ember-state-services/blob/master/addon/state-for.js#L79-L81). Once the object has been created initialState is never called again and any future calls to initialState wont change previous objects state. InitialState should be thought of as a pure function and I believe the funkiness that @stefanpenner is describing is that I set the context to something instead of passing the environment into this function.